storagemarket/impl/client_environments.go

```diff
@@ -65,9 +66,20 @@ func (csg *clientStoreGetter) Get(proposalCid cid.Cid) (bstore.Blockstore, error
 		return nil, xerrors.Errorf("failed to get client deal state, err=%w", err)
 	}
 
-	// get a read Only CARv2 blockstore that provides random access on top of the client's CARv2 file containing the CARv1 payload
-	// that needs to be transferred as part of the deal.
-	return csg.c.readOnlyCARStoreTracker.GetOrCreate(proposalCid.String(), deal.CARv2FilePath)
+	// get a read Only CARv2 blockstore that provides random access on top of
+	// the client's CARv2 file containing the CARv1 payload that needs to be
+	// transferred as part of the deal.
+	bs, err := csg.c.readOnlyCARStoreTracker.GetOrCreate(proposalCid.String(), deal.CARv2FilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Wrap the ReadBlockstore with a Blockstore implementation that panics
+	// if any write methods are called (eg Put, Delete).
+	// We need to do this because the Get interface returns a
```

Comment:
The `ReadOnlyBlockstore` returned by the CARv2 lib takes care of this and we can get rid of this.

Reply:
Makes sense to get rid of it here 👍  I've now done that.

We can't get rid of it entirely however as the ShardAccessor returns a Blockstore of type ReadBlockstore so we need to wrap it.